### DISCUSSION TITLE
Ad hoc filter: Support using data source variables

### DISCUSSION THREAD

**Discussion by arnedc** [grafana/grafana#37033] [Dashboards, Panels & Visualizations]
**What happened**: Added the data source variable and an add hoc filter. When selecting a custom data source, the filter gets applied. However, when selecting default as data source, the filter doesn't get applied.
In the images the datasource variable is named 'DATASOURCE' with label 'Location' and DCG is the default Datasource:
![Datasource_variable](https://user-images.githubusercontent.com/8348681/87138454-bbd2d800-c29e-11ea-9a70-65f06bdcbe06.png)
![Filters_non-default_datasource](https://user-images.githubusercontent.com/8348681/87138160-4ebf4280-c29e-11ea-8af6-d0f6da33b8d8.png)
![Filters_default_datasource](https://user-images.githubusercontent.com/8348681/87138178-57177d80-c29e-11ea-8749-a4ed1f7395fa.png)

**What you expected to happen**: Ad hoc filters get applied even when default data source is applied

**How to reproduce it (as minimally and precisely as possible)**: 
* Set a Prometheus data source as default
* Create a datasource variable
* Create a panel which queries the data source variable
* Apply an ad hoc filter
* Watch the difference when selecting the default data source or the specific data source as the datasource variable.

**Anything else we need to know?**:
This does seem to work when querying the default data source immediately (without defining a datasource variable)

**Environment**:
- Grafana version: 6.7.3
- Data source type & version: Prometheus 2.19.0
- OS Grafana is installed on: Ubuntu 18.04.2 LTS
- User OS & Browser: Windows / Chrome
- Grafana plugins: none
- Others: datasources and dashboards are provisioned at set up

**Comment by torkelo** [comment]
Can you show the ad-hoc filter data source settings?

**Comment by arnedc** [comment]
Here you go, it also uses the DATASOURCE variable
![Filter_variable](https://user-images.githubusercontent.com/8348681/87151071-7d93e380-c2b3-11ea-81ad-48af07827e99.png)

**Comment by aocenas** [comment]
Reproduced also on v7.2.0-pre (7cdbae4ae6). Only happens if the datasource is the `default` datasource.

**Comment by torkelo** [comment]
Changing this to a feature request to support data source variables with ad-hoc variables

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
